core/registry.js

```diff
@@ -273,6 +278,31 @@ Blockly.registry.getObject = function(type, name, opt_throwIfMissing) {
     Blockly.registry.getItem_(type, name, opt_throwIfMissing));
 };
 
+/**
+ * Returns a map of items registered with the given type.
+ * @param {string|!Blocklly.registry.Type<T>} type The type of the plugin.
+ *     (e.g. Category)
+ * @param {boolean=} opt_throwIfMissing Whether or not to throw an error if we
+ *     are unable to find the object.
+ * @return {?Object<string, ?T|?function(new:T, ...?)>} A map of objects with
+ *     the given type, or null if none exists.
+ * @template T
+ */
+Blockly.registry.getAllItems = function(type, opt_throwIfMissing) {
+  type = String(type).toLowerCase();
+  var typeRegistry = Blockly.registry.typeMap_[type];
```

Comment:
Hmm there's a bit of a problem with this. When you register an item the case is removed (not sure why), so all of the keys in this map are caseless. This is incorrect for the JSON serializer because according to the [styleguide](https://google.github.io/styleguide/jsoncstyleguide.xml#Property_Name_Format), key names should be camelCased.

We could:
  1) Stop removing the case from keys.
  2) Keep a separate map the maintains casing.
  3) Require developers to supply the names of their serializers twice. Once in registration, and once in the definition of their serializer.

Reply:
> I think one of the reasons we did this was that it slightly decreases the chance of mistakes for developers who have to type in the plugin name to the options struct.

Ah that makes sense. And I guess it also means that we can't go with 1, without breaking people who *are* accessing registered items using the caseless name. And we don't even throw errors by default, so no help there.

But I'm not sure if breaking people here is a big deal or not.

Reply:
I'm not sure it is a big deal. I think probably the subset of people that are using this feature and that have incorrectly capitalized is probably pretty small. We should add it to the release notes since it is a change in the public API, but I don't think it is worth complicating the code (2) or adding extra work for the developer (3).

Reply:
So I tried to implement (1) and I don't think it's going to be workable :/ The issue is that the toolbox uses tag names from XML to access things in the registry, and we can't rely on the case of tag names due to IE ([relevant code](https://github.com/google/blockly/blob/f94b1db85018b5a7261fa01f7d04cd5c8367fe2c/core/utils/toolbox.js#L345)).

So given that, I'm going to attempt to implement (2) instead.

Reply:
Got (2) working!